SU2_CFD/src/solver_direct_heat.cpp

```diff
@@ -578,3 +578,1978 @@ void CHeatSolver::ImplicitEuler_Iteration(CGeometry *geometry, CSolver **solver_
   SetResidual_RMS(geometry, config);
   
 }
+
+CHeatSolverFVM::CHeatSolverFVM(void) : CSolver() {
+
+  ConjugateVar = NULL;
+}
+
+CHeatSolverFVM::CHeatSolverFVM(CGeometry *geometry, CConfig *config, unsigned short iMesh) : CSolver() {
+
+  unsigned short iVar, iDim, nLineLets, iMarker;
+  unsigned long iPoint, index, iVertex;
+  su2double dull_val;
+
+  unsigned short iZone = config->GetiZone();
+  unsigned short nZone = geometry->GetnZone();
+  bool restart = (config->GetRestart() || config->GetRestart_Flow());
+  bool adjoint = (config->GetContinuous_Adjoint()) || (config->GetDiscrete_Adjoint());
+  bool dual_time = ((config->GetUnsteady_Simulation() == DT_STEPPING_1ST) ||
+                    (config->GetUnsteady_Simulation() == DT_STEPPING_2ND));
+  bool time_stepping = config->GetUnsteady_Simulation() == TIME_STEPPING;
+
+  int rank = MASTER_NODE;
+
+  bool flow = ((config->GetKind_Solver() == NAVIER_STOKES)
+               || (config->GetKind_Solver() == RANS)
+               || (config->GetKind_Solver() == DISC_ADJ_NAVIER_STOKES)
+               || (config->GetKind_Solver() == DISC_ADJ_RANS));
+  bool compressible = config->GetKind_Regime() == COMPRESSIBLE;
+  bool heat_equation      = config->GetKind_Solver() == HEAT_EQUATION_FVM;
+  unsigned short turbulent = config->GetKind_Turb_Model();
+
+#ifdef HAVE_MPI
+  MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+#endif
+
+  /*--- Dimension of the problem --> temperature is the only conservative variable ---*/
+
+  nVar = 1;
+  nPoint = geometry->GetnPoint();
+  nPointDomain = geometry->GetnPointDomain();
+
+  /*--- Initialize nVarGrad for deallocation ---*/
+
+  nVarGrad = nVar;
+
+  /*--- Define geometry constants in the solver structure ---*/
+
+  nDim = geometry->GetnDim();
+  node = new CVariable*[nPoint];
+  nMarker = config->GetnMarker_All();
+
+  CurrentMesh = iMesh;
```

Comment:
Is multigrid working well for the new FVM heat solver?

Reply:
It did in all in cases where it worked for the flow solver, too.
So I like to think that all failing cases were due to the general partitioning problem.

As a interim solution, only a CSingleGridIntegration is used which still performs well.
(But I saw that you just opened up a topic on that matter, so maybe I can make the multigrid usage up to a user preference soon!)